File: Yona/Yona/Models/TileDraft.swift
//
//  TileDraft.swift
//  Yona
//
//  The editable fields of a tile, bundled together for create/update so the
//  repository and store don't pass them around as a long parameter list.
//

import Foundation

struct TileDraft {
    var title: String
    var url: String
    var notes: String?
    var costAmount: Double?
    var costPeriod: CostPeriod?
}

File: android/app/src/main/java/com/yona/app/core/TileDraft.kt
package com.yona.app.core

/**
 * The editable fields for creating/updating a tile (mirrors the iOS TileDraft).
 * Dates are ISO "yyyy-MM-dd" strings, matching the Postgres `date` column.
 */
data class TileDraft(
    val title: String,
    val url: String,
    val notes: String?,
    val costAmount: Double? = null,
    val costPeriod: String? = null,
    val renewalDate: String? = null,
    val renewalRepeat: String? = null,
    val billingSource: String? = null,
    val paymentMethod: String? = null,
)
